### PR TITLE
ConditionConverter query parameter conversion

### PR DESCRIPTION
**Pull request by ilaflamme-va**
https://vajira.max.gov/browse/API-20110

Mappings
```
R4                      ---->           DSTU2
-----------------------------------------------
encounter-diagnosis     ---->           diagnosis
problem-list-item       ---->           problem
health-concern          ---->           health-concern
```

### CODE REVIEW

File: fhir-resource-conversion/src/main/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/ConditionConverter.java [hunk 1]
```diff
@@ -186,4 +192,37 @@ static String toR4(String dstu2) {
...
+      if (isBlank(source)) {
+        return null;
+      }
+      switch (source) {
+        case "problem":
+          return "problem-list-item";
+        case "diagnosis":
+          return "encounter-diagnosis";
+        default:
+          return source;
```
Comment (KevinCranmer) [inline fhir-resource-conversion/src/main/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/ConditionConverter.java:224]: If the dstu2 value being converted is `complaint` then the R4 values would also be `complaint` from this method - which looks like it should be `health-concern`(?) based on the comment on line 197.

Comment (ilaflamme-va) [reply fhir-resource-conversion/src/main/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/ConditionConverter.java:224]: Oh you're right. I missed a couple conversion cases here!

File: fhir-resource-conversion/src/main/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/ConditionConverter.java [hunk 2]
```diff
@@ -186,4 +192,43 @@ static String toR4(String dstu2) {
...
+      switch (source) {
+        case "problem":
+          return "problem-list-item";
+        case "diagnosis":
+        case "finding":
+          return "encounter-diagnosis";
+        case "complaint":
+        case "symptom":
+          return "health-concern";
+        case "":
```
Comment (jhulbert-va) [inline fhir-resource-conversion/src/main/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/ConditionConverter.java:228]: Why add a specific case for `""` and falling through to default?

File: fhir-resource-conversion/src/test/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/ConditionConverterTest.java [hunk 3]
```diff
@@ -141,9 +150,7 @@ void nullValueReturnsNull() {
   }
 
   @ParameterizedTest
-  @CsvSource(textBlock = """
-    clinical-status,active,clinicalstatus,active
-    """)
+  @MethodSource
```
Comment (jhulbert-va) [inline fhir-resource-conversion/src/test/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/ConditionConverterTest.java:153]: Why switch from csv source to method source?

File: fhir-resource-conversion/src/test/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/ConditionConverterTest.java [hunk 4]
```diff
@@ -152,6 +159,19 @@ void queryParameterConverter(String r4name, String r4value, String dstu2Name, St
     assertThat(qpc.inverse().copyAndConvert(dstu2)).usingRecursiveComparison().isEqualTo(r4);
   }
 
+  @ParameterizedTest
+  @CsvSource(
+      textBlock =
+          """
+    ,
```
Comment (jhulbert-va) [inline fhir-resource-conversion/src/test/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/ConditionConverterTest.java:166]: I tend to prefer using csv source's `nullValue` field to set null to represent null. I think it's more explicit and reads better.

File: fhir-resource-conversion/src/test/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/ConditionConverterTest.java [hunk 5]
```diff
@@ -168,6 +188,22 @@ void toDstu2ClinicalStatusCodeValue(String r4, String dstu2) {
     assertThat(ClinicalStatusMapping.toDstu2(r4)).isEqualTo(dstu2);
   }
 
+  @ParameterizedTest
+  @CsvSource(
+      textBlock =
+          """
+    ,
```
Comment (jhulbert-va) [inline fhir-resource-conversion/src/test/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/ConditionConverterTest.java:195]: Same here.

File: fhir-resource-conversion/src/test/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/ConditionConverterTest.java (hunk 3 above)
Comment (ilaflamme-va) [reply fhir-resource-conversion/src/test/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/ConditionConverterTest.java:153]: reads better with 4 params

**Review by jhulbert-va — APPROVED** [review]

**Review by jblefkove-va — APPROVED** [review]

**Pull request merged**